src/main/java/de/tum/in/www1/artemis/web/rest/admin/AdminCourseResource.java

```diff
@@ -102,9 +108,11 @@ public ResponseEntity<Course> createCourse(@RequestPart Course course, @RequestP
             course.setCourseIcon(pathString);
         }
 
-        Course result = courseRepository.save(course);
+        Course createdCourse = courseRepository.save(course);
+
+        Arrays.stream(DefaultChannelType.values()).toList().forEach(channelType -> this.createChannel(createdCourse, channelType));
```

Comment:
What's the reason we need the ``toList()``? I'm not quite getting it tbh. For me it compiles without it, too.
Can't read much from the conversation above ending with ``A: Yes`` 🤔